### ISSUE TITLE
Roberta (and BERT) tokenization converts "do not" to "don't"

### ISSUE THREAD

**Issue by davisyoshida** [huggingface/transformers#3846]
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...): Roberta

Language I am using the model on (English, Chinese ...): English

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [x] my own task or dataset: (give details below)

## To reproduce

Execute the following:
```python
import transformers

tokenizer = transformers.RobertaTokenizer.from_pretrained('roberta-base')
print(tokenizer.decode(tokenizer.encode('is not')))
print(tokenizer.decode(tokenizer.encode('do not')))
```

The output is
```
<s> don't</s>
<s> is not</s>
```

## Expected behavior
The detokenization should not incorrectly introduce a contraction.

## Environment info

     
- `transformers` version: 2.8.0
- Platform: Linux-5.5.3-arch1-1-x86_64-with-glibc2.2.5
- Python version: 3.8.1
- PyTorch version (GPU?): not installed (NA)
- Tensorflow version (GPU?): not installed (NA)
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

**Comment by davisyoshida** [comment]
For anyone coming across this issue, you can disable all such transformations in `decode` by passing `clean_up_tokenization_spaces=False`. However, I maintain that this decoding behavior is not a sensible default.

**Comment by patrickvonplaten** [comment]
fixed with #4024
